### PR TITLE
Upgrade CI binary build runner from 4x to 12xlarge

### PR DESCRIPTION
**Pull request by huydhn**
It currently takes a whopping 2h30m just to build PyTorch binary for every PR and commit. Pushing it to 12xlarge reduces the time to 1h40m https://github.com/pytorch/pytorch/actions/runs/3323869550/jobs/5494754029, not exactly a linear (and fair) trade, but good enough to reduce this long pole.

I'll monitor the queue for 12xlarge after this change.

### CODE REVIEW

**Review by kit1980 — APPROVED** [review]

**Review by malfet — APPROVED** [review]

File: .github/workflows/_binary-build-linux.yml [hunk 1]
```diff
@@ -67,7 +67,7 @@ on:
 
 jobs:
   build:
-    runs-on: linux.4xlarge
+    runs-on: linux.12xlarge
     timeout-minutes: 270
```
Comment (malfet) [inline .github/workflows/_binary-build-linux.yml:71]: Should we reduce timeout minutes to 120 for example?

Comment (huydhn) [reply .github/workflows/_binary-build-linux.yml:71]: Sure, we can do so

**Comment by seemethere** [comment]
Another solution is to just potentially limit the amount of architectures we build for, especially for PRs

**Comment by huydhn** [comment]
> Another solution is to just potentially limit the amount of architectures we build for, especially for PRs

From what I see, we only build it for cuda 11.6 and python 3.7 here:

```
      GPU_ARCH_VERSION: 11.6
      GPU_ARCH_TYPE: cuda
      DOCKER_IMAGE: pytorch/manylinux-builder:cuda11.6
      DESIRED_PYTHON: "3.7"
```

Do you mean using something like `TORCH_CUDA_ARCH_LIST=5.2`?  I wonder what is the implication of this. I learn new lessons today about many different requirements between binary build v.s. CI build :)

**Comment by ZainRizvi** [comment]
Do we want to leave the timeout at some higher value? Setting it to just 10 mins above the expected build time doesn't seem like it leaves much margin

**Pull request closed without merging**